### DISCUSSION TITLE
[RR2] Question: Using multiple workers and supervisor

### DISCUSSION THREAD

**Discussion by jwillp** [roadrunner-server/roadrunner#632] [Q&A] [answered]
Hi,

I haven't seen anything in the documentation regarding this, but is it possible to specify different workers that can be supervised and automatically restarted?

For example:
Let's say that I have in addition to the http server, the need to run a few symfony console commands as background processes in a long running fashion that perform some specific tasks. What is the preferred way of doing this?

Can I somehow specify these commands to be supervised and restarted when they stop, using the `.rr.yaml` file, or should I define multiple `.rr.yaml` files for each of these commands and run multiple times the `rr serve` with each one of these configurations?

Also, how would I specify this if they don't actually need to be served using http?

Thank you

**Comment by rustatian** [comment, marked as answer]
Hey @jwillp . At the moment RR2 supports only 1 `server` inside it with 1 command to start workers. I guess you can create a simple bash script that runs all your RR2 instances with the different `.rr.yaml`s and commands inside them.

> Also, how would I specify this if they don't actually need to be served using http?

We discussed this question, and yeah, we have a plan on our radar to create a new plugin, which would execute PHP script w/o HTTP endpoint, but like background processes. But at the moment, you have to use HTTP plugin for that.

**Reply by rustatian** [reply]
@jwillp I hope, that I will be able to make this plugin in time for the `2.0.5` release [2021-04-27] https://github.com/spiral/roadrunner/issues/633

**Reply by jwillp** [reply]
Hi @48d90782  thank you for the quick response. Will be looking forward to that plugin then!
For launching multiple services I guess I could go with something like https://github.com/struCoder/pmgo to manage these processes.

**Reply by rustatian** [reply]
Yeah, looks pretty good :)
